test/Dialect/FIRRTL/imconstprop.mlir

```diff
@@ -224,25 +224,46 @@ firrtl.circuit "testDontTouch"  {
 }
 
 firrtl.circuit "OutPortTop" {
-    firrtl.module @OutPortChild1(out %out: !firrtl.uint<1>) attributes {
-      portAnnotations =[[{class = "firrtl.transforms.DontTouchAnnotation"}]]} {
-      %c0_ui1 = firrtl.constant 0 : !firrtl.uint<1>
-      firrtl.connect %out, %c0_ui1 : !firrtl.uint<1>, !firrtl.uint<1>
-    }
-    firrtl.module @OutPortChild2(out %out: !firrtl.uint<1>) {
-      %c0_ui1 = firrtl.constant 0 : !firrtl.uint<1>
-      firrtl.connect %out, %c0_ui1 : !firrtl.uint<1>, !firrtl.uint<1>
-    }
+  firrtl.module @OutPortChild1(out %out: !firrtl.uint<1> {firrtl.annotations = [{class = "firrtl.transforms.DontTouchAnnotation"}]}) {
+    %c0_ui1 = firrtl.constant 0 : !firrtl.uint<1>
+    firrtl.connect %out, %c0_ui1 : !firrtl.uint<1>, !firrtl.uint<1>
+  }
+  firrtl.module @OutPortChild2(out %out: !firrtl.uint<1>) {
+    %c0_ui1 = firrtl.constant 0 : !firrtl.uint<1>
+    firrtl.connect %out, %c0_ui1 : !firrtl.uint<1>, !firrtl.uint<1>
+  }
   // CHECK-LABEL: firrtl.module @OutPortTop
-    firrtl.module @OutPortTop(in %x: !firrtl.uint<1>, out %z: !firrtl.uint<1>) {
-      %c_out = firrtl.instance @OutPortChild1  {name = "c"} : !firrtl.uint<1>
-      %c_out_0 = firrtl.instance @OutPortChild2  {name = "c"} : !firrtl.uint<1>
-      // CHECK: %0 = firrtl.and %x, %c_out
-      %0 = firrtl.and %x, %c_out : (!firrtl.uint<1>, !firrtl.uint<1>) -> !firrtl.uint<1>
-      // CHECK: %1 = firrtl.and %0, %c0_ui1
```

Comment:
With this change this was being optimized away `and x, 0 -> 0`.  Since we are marking the users of the port as overdefined (instead of nothing), they are now being added to the worklist and the folder gets a chance to run (i think..).

Reply:
Interesting... 🤔  The simpler test this is changed to looks fine by me (and seems to be a simpler way to test the same effect).